apps/src/weblab/WebLab.js

```diff
@@ -419,6 +437,11 @@ WebLab.prototype.setBrambleHost = function (obj) {
     if (this.hideSource) {
       this.brambleHost.enableFullscreenPreview();
     }
+    // Enable the Finish/Submit/Unsubmit button if it is present:
+    let shareCell = document.getElementById('share-cell');
+    if (shareCell) {
+      shareCell.className = 'share-cell-enabled';
+    }
```

Comment:
would be great to get this logic into React, but it looks like this is following the pattern currently used in AppLab, so probably makes sense for you to just emulate that for now.